### ISSUE TITLE
LLVM 23 raw profiles (version 11) misparse: header, ProfileData record, and tail padding all changed

### ISSUE THREAD

**Issue by Nexlab-One** [xd009642/llvm-profparser#81]
## Summary

`.profraw` files produced by a toolchain built against **LLVM 23** are misparsed. LLVM 23 bumped `INSTR_PROF_RAW_VERSION` to 11 and changed **two** structures — the raw header and the per-function `ProfileData` record — and it also moved where the struct's tail padding lives. The visible symptom is `parser failure: Nom(Satisfy)`, preceded by `consistency check for reading counts failed`.

This currently breaks `cargo tarpaulin --engine llvm` on current Rust nightly for everyone, not just us.

I have a working, end-to-end-tested fix and will open a PR alongside this issue.

## Steps to reproduce

A three-line crate, no dependencies:

```rust
pub fn add(a: u32, b: u32) -> u32 { a + b }
#[cfg(test)]
mod tests { #[test] fn t() { assert_eq!(super::add(2,2), 4); } }
```

```
cargo +nightly tarpaulin --engine llvm
```

## Expected vs actual

* **Expected:** coverage is reported, as it is on stable.
* **Actual:** the test itself compiles and passes, then parsing fails:

```
running 1 test
test tests::t ... ok
test result: ok. 1 passed; 0 failed

ERROR llvm_profparser::instrumentation_profile::raw_profile: consistency check for reading counts failed
ERROR cargo_tarpaulin: parser failure: Nom(Satisfy)
Error: "parser failure: Nom(Satisfy)"
```

## Root cause

Diffing `llvm/include/llvm/ProfileData/InstrProfData.inc` between `release/22.x` and `release/23.x`:

**1. `INSTR_PROF_RAW_VERSION` 10 → 11.**

**2. Three new `uint64_t` fields in the middle of the raw header**, before `NamesSize`:

```diff
 INSTR_PROF_RAW_HEADER(uint64_t, PaddingBytesAfterBitmapBytes, PaddingBytesAfterBitmapBytes)
+INSTR_PROF_RAW_HEADER(uint64_t, NumUniformCounters, NumUniformCounters)
+INSTR_PROF_RAW_HEADER(uint64_t, PaddingBytesAfterUniformCounters, PaddingBytesAfterUniformCounters)
+INSTR_PROF_RAW_HEADER(uint64_t, UniformCountersDelta, ...)
 INSTR_PROF_RAW_HEADER(uint64_t, NamesSize,  NamesSize)
```

`parse_header` gates on `version >= 7`, `>= 9` and `>= 10`, so with no `>= 11` branch it reads straight past them and every later field is shifted by 24 bytes. Critically `counters_delta` then holds `PaddingBytesAfterUniformCounters`, and `counters_delta` is what `read_raw_counts` uses for its offset arithmetic — hence the consistency-check failure.

**3. Two new fields in the per-function `ProfileData` record:**

```diff
 INSTR_PROF_DATA(const IntPtrT, IntPtrTy, CounterPtr, RelativeCounterPtr)
+INSTR_PROF_DATA(const IntPtrT, IntPtrTy, UniformCounterPtr, RelativeUniformCounterPtr)
 INSTR_PROF_DATA(const IntPtrT, IntPtrTy, BitmapPtr, RelativeBitmapPtr)
...
 INSTR_PROF_DATA(const uint16_t, Int16ArrayTy, NumValueSites[IPVK_Last+1], ...)
+INSTR_PROF_DATA(const uint16_t, llvm::Type::getInt16Ty(Ctx), OffloadDeviceWaveSize, ...)
 INSTR_PROF_DATA(const uint32_t, llvm::Type::getInt32Ty(Ctx), NumBitmapBytes, ...)
```

**4. The struct's tail padding moved, which is the part that is easy to miss.** In v9/v10, `ProfileData` ends on a 4-byte field at offset 60, so C pads the struct to 64 — which is what this line in `parse_bytes` is compensating for:

```rust
if version_num > 8 {
    let (bytes, v) = take(4usize)(bytes)?; // TODO WHAT AM I MISSING HERE?
```

(That `TODO` is answered: it is struct tail padding.)

In v11, `OffloadDeviceWaveSize` pushes `NumBitmapBytes` to offset 68 and the struct ends at 72, which is already 8-aligned — so there is **no** tail padding, and the 2 bytes of padding move *inside* the struct, between `OffloadDeviceWaveSize` and `NumBitmapBytes`. Taking 4 bytes as well over-consumes and desynchronises every record after the first.

I found this empirically: after adding only the header and record fields, record 0 parsed correctly and record 1 started at byte 260 instead of 256.

## Evidence

Profraw headers from the same crate on two toolchains:

```
nightly: 8172 666f 7270 6cff 0b00 ...   -> version 0x0b (11)
stable:  8172 666f 7270 6cff 0a00 ...   -> version 0x0a (10)
```

| | rustc | LLVM | profraw version | result |
|---|---|---|---|---|
| stable | 1.96.0 (ac68faa20) | 22.1.2 | 10 | parses |
| nightly | 1.99.0-nightly (771916f90 2026-08-08) | 23.1.0 | 11 | fails |

`--engine ptrace` succeeds on the same nightly, so the instrumented binary and the test run are fine; only profile parsing fails.

With the three-part fix applied, the full v11 header and both `ProfileData` records decode correctly (`NumCounters` 2 and 1, summing to the header's `counters_len` of 3), and end-to-end:

```
cargo +nightly tarpaulin --engine llvm --out Stdout
100.00% coverage, 1/1 lines covered
```

which matches what the `ptrace` engine independently reports for the same crate.

## Environment

| Field | Value |
|---|---|
| OS | Linux 6.6.114.1 (WSL2), x86_64 |
| Target | `x86_64-unknown-linux-gnu` |
| llvm-profparser | master @ cc5c8c9 |
| cargo-tarpaulin | 0.37.0 (installed `--locked`) |

## Not checked

* Whether the same failure occurs on macOS or Windows, or on non-x86_64 targets. Linux x86_64 only.
* Whether any LLVM release between 22 and 23 is also affected — only the two endpoints were tested.
* **Whether `UniformCounterPtr` / `NumUniformCounters` / `UniformCountersDelta` need to be USED for anything beyond correct offsets.** I only established that reading them realigns parsing; if LLVM 23 emits uniform counters in some configuration, they may carry data that should be surfaced rather than skipped. The names suggest a new counter representation, so this is worth a second opinion from someone who knows the LLVM side.
* Likewise `OffloadDeviceWaveSize` is read and stored but otherwise unused by the fix.
* I did not add a v11 `.profraw` fixture to the test suite; the repo's existing fixtures are pre-v11 and I did not want to guess at your preferred way of generating and committing one.
